infra.ci.jenkins.io.tf

```diff
@@ -76,7 +76,7 @@ module "infra_ci_jenkins_io_fileshare_serviceprincipal_writer" {
   active_directory_owners    = [data.azuread_service_principal.terraform_production.id]
   active_directory_url       = "https://github.com/jenkins-infra/azure"
   service_principal_end_date = "2024-06-20T23:00:00Z"
-  file_share_id              = azurerm_storage_share.contributors_jenkins_io.resource_manager_id
+  file_share_id              = azurerm_resource_group.contributors_jenkins_io.id
```

Comment:
Scoping to resource group works and is ok for the time being but it looks like it is not restricting enough. 

WDYT about trying with a restriction at the SA level (in-between the filestorage and the resource group).

If you haven't tried, I don't mind checking later once you'll have set up this system (and I'll add a comment here subsequently)